source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -140,6 +128,9 @@ void EnvoyQuicServerStream::resetStream(Http::StreamResetReason reason) {
   } else {
     Reset(envoyResetReasonToQuicRstError(reason));
   }
+  if (buffer_memory_account_) {
+    buffer_memory_account_->clearDownstream();
+  }
```

Comment:
Consider doing this before reset callbacks.

Reply:
Good idea, done.